listenstore/listenstore/listen.py

```diff
@@ -9,7 +9,7 @@ def __init__(self, user_id=None, timestamp=None, artist_msid=None, album_msid=No
                  recording_msid=None, data=None):
         self.user_id = user_id
         self.timestamp = timestamp
-        self.ts_since_epoch = calendar.timegm(self.timestamp.utctimetuple())
+        self.ts_since_epoch = calendar.timegm(self.timestamp.utctimetuple()) if self.timestamp else None
```

Comment:
The redis-consumer calls `Listen().from_json()` which needs to initialize the class before making the call to method, as its a `classmethod`. So i had 2 options, either to make it a staticmethod or do the above change.
Also when dealing with `playing_now` listen, they cannot have a timestamp. Currently they have current timestamp (i do not use this anywhere). But ideally their timestamps should be None. And they have an additional `playing_now` attribute to be true.